content/en/docs/concepts/services-networking/service.md

```diff
@@ -91,22 +119,22 @@ spec:
       targetPort: 9376
 ```
 
-This specification creates a new Service object named "my-service", which
-targets TCP port 9376 on any Pod with the `app.kubernetes.io/name=MyApp` label.
+That manifest creates a new Service named "my-service", which
+targets TCP port 9376 on any Pod with the `app.kubernetes.io/name: MyApp` label.
 
 Kubernetes assigns this Service an IP address (sometimes called the "cluster IP"),
 which is used by the Service proxies
 (see [Virtual IPs and service proxies](#virtual-ips-and-service-proxies) below).
 
-The controller for the Service selector continuously scans for Pods that
-match its selector, and then POSTs any updates to an Endpoint object
-also named "my-service".
+The {{<glossary_tooltip text="controller" term_id="controller">}} for the Service
+continuously scans for Pods that match its selector, and then POSTs any updates to
+a set of [EndpointSlices](#endpointslices) that together represent the backend Pods.
 
-{{< note >}}
-A Service can map _any_ incoming `port` to a `targetPort`. By default and
-for convenience, the `targetPort` is set to the same value as the `port`
-field.
-{{< /note >}}
+The Service controller also manages an older API, [Endpoints](#endpoints), that is
+similar to EndpointSlice (but has some limitations around scalability). The Endpoints
+API is widely used and remains a supported Kubernetes API.
```

Comment:
there is no service controller, is the endpoints or endpointslice controller that does the mapping based on the selector